core/src/main/java/org/elasticsearch/rest/action/search/RestSearchAction.java

```diff
@@ -130,15 +130,12 @@ private static void parseSearchSource(final SearchSourceBuilder searchSourceBuil
             searchSourceBuilder.query(queryBuilder);
         }
 
-        int from = request.paramAsInt("from", -1);
-        if (from != -1) {
-            searchSourceBuilder.from(from);
+        if (request.hasParam("from")) {
```

Comment:
I would be okay with rejecting anything negative here, on the REST layer. We still need a way to represent unset internally. Using `null` introduces boxing, we should avoid that. Let's continue to use -1, that's the idiom.

However, I would like to solicit feedback from others before you proceed with rejecting anything negative including -1 here. What do you think @javanna?